src/test/unit/model/array_functions_roundtrip_test.cpp

```diff
@@ -0,0 +1,91 @@
+#include <stan/model/log_prob_grad.hpp>
+#include <stan/io/empty_var_context.hpp>
+#include <stan/io/json/json_data.hpp>
+#include <test/test-models/good/model/parameters.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+class ModelArrayFunctionsRoundtripTest : public testing::Test {
+ public:
+  ModelArrayFunctionsRoundtripTest()
+      : model(context, 0, nullptr), rng(12324232), inits(nullptr) {
+    out.str("");
+
+    std::vector<std::string> json_path = {
+        "src", "test", "test-models", "good", "model", "parameters.inits.json"};
+    std::string filename = paths_to_fname(json_path);
+    std::ifstream in(filename);
+    inits = new stan::json::json_data(in);
+  }
+
+  ~ModelArrayFunctionsRoundtripTest() { delete inits; }
```

Comment:
I would do something like this so the deletion just gets handled by a `unique_ptr`

```suggestion
auto init_inits() {
  std::vector<std::string> json_path = {
    "src", "test", "test-models", "good", "model", "parameters.inits.json"};
  std::string filename = paths_to_fname(json_path);

  std::fstream stream(filename, std::fstream::in);
  return std::unique_ptr<stan::io::var_context>(new stan::json::json_data(stream));
}

class ModelArrayFunctionsRoundtripTest : public testing::Test {
 public:
  ModelArrayFunctionsRoundtripTest()
      : model(context, 0, nullptr), rng(12324232), inits(init_inits()) {}

  void TearDown() override {
    out.str("");
  }


```